tests/test_request_dict.py

```diff
@@ -130,7 +138,22 @@ def parse(self, response):
         spider = MySpider()
         r = Request("http://www.example.com", callback=spider.parse)
         setattr(spider, 'parse', None)
-        self.assertRaises(ValueError, request_to_dict, r, spider=spider)
+        self.assertRaises(ValueError, r.to_dict, spider=spider)
+
+    def test_callback_not_available(self):
+        """Callback method is not available in the spider passed to from_dict"""
+        spider = TestSpiderDelegation()
+        r = Request("http://www.example.com", callback=spider.delegated_callback)
+        d = r.to_dict(spider=spider)
+        self.assertRaises(ValueError, request_from_dict, d, spider=Spider("foo"))
+
+
+class DeprecatedMethodsRequestSerializationTest(RequestSerializationTest):
+    def _assert_serializes_ok(self, request, spider=None):
+        from scrapy.utils.reqser import request_from_dict as _request_from_dict, request_to_dict as _request_to_dict
```

Comment:
I’m not sure it’s worth it, but for completeness maybe we could catch the deprecation warning logged by this import, just for the sake of verifying it’s there.

Reply:
I actually attempted to do so, but I kept having issues with the warnings not being reset and appearing only once, which made only the first check succeed. Consider the following:
```python
Python 3.8.2 (default, Apr 18 2020, 17:39:30) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.22.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import warnings

In [2]: warnings.simplefilter("always")

In [3]: from scrapy.utils.reqser import request_to_dict
<ipython-input-3-75e30ca4f2be>:1: ScrapyDeprecationWarning: Module scrapy.utils.reqser is deprecated, please use scrapy.Request.to_dict  and/or scrapy.utils.request.request_from_dict instead
  from scrapy.utils.reqser import request_to_dict

In [4]: warnings.resetwarnings()

In [5]: del request_to_dict

In [6]: from scrapy.utils.reqser import request_to_dict  # no warning here

In [7]: 
```

It seems like this is caused by the warning being raised at module level. It does work if it's deleted with `del sys.modules["scrapy.utils.reqser"]`, it should be safe within a test.